### PR TITLE
Extract Raven::Instance and delegate from Raven to it

### LINKED ISSUE #462: Multiple Raven Instances?
Is there a way to configure multiple Raven instances? For example: sending a special subset of errors to one DSN and most errors to another DSN.

### PR DESCRIPTION
**Pull request by phillbaker**
This refactor was done in 2 steps:

1. Extract methods from Raven's top level module (in base.rb) into Raven::Instance, copy the tests, get them green
1. Delegate from `Raven` to an instance of `Raven::Instance`

These instances may be used like this:
```
class Foo
  def initialize
    @other_raven = Raven::Instance.new
    @other_raven.configure do |config|
      config.server = 'http://...'
    end
  end

  def foo
    ...
  rescue => e
    @other_raven.capture_exception(e)
  end
end
```

Fixes #462.

### CODE REVIEW

**Comment by phillbaker** [comment]
@nateberkopec @dcramer what do you think?

**Comment by nateberkopec** [comment]
@phillbaker Thanks for getting this running against Rubocop. I'll take a look as soon as I can.

File: lib/raven/instance.rb [hunk 1]
```diff
@@ -0,0 +1,219 @@
+require 'raven/base'
+require 'English'
```
Comment (nateberkopec) [inline lib/raven/instance.rb]: ?

File: lib/raven/instance.rb [hunk 2]
```diff
@@ -58,7 +67,6 @@ def report_status
         logger.info "Raven #{VERSION} configured not to send errors."
       end
     end
-    alias_method :report_ready, :report_status
```
Comment (nateberkopec) [inline lib/raven/instance.rb]: Why remove these? It seems like it only complicates your `Forwardable` related code.

File: lib/raven/instance.rb [hunk 3]
```diff
@@ -90,13 +98,17 @@ def send_event(event)
     #     MyApp.run
     #   end
     def capture(options = {})
```
Comment (nateberkopec) [inline lib/raven/instance.rb]: Why was this changed?

**Comment by nateberkopec** [comment]
Could we alias `Raven.client` to `Raven.instance`?

File: lib/raven/instance.rb (hunk 1 above)
Comment (phillbaker) [inline lib/raven/instance.rb]: This was from rubocop - which request that `$!` on L211 be replaced with $ERROR_INFO from this module.

File: lib/raven/instance.rb (hunk 2 above)
Comment (phillbaker) [inline lib/raven/instance.rb]: `Forwardable` doesn't support forwarding to aliases, (but you can set up an alias at the top level that forwards to the same underlying method) so all the aliases were moved up to `Base`. That also kept all the aliases consolidated, which seemed a bit cleaner if more needed to be added in the future.

File: lib/raven/base.rb [hunk 4]
```diff
@@ -241,12 +87,6 @@ def rails_safely_prepend(module_name, opts = {})
...
 
-    # For cross-language compat
-    alias :captureException :capture_exception
-    alias :captureMessage :capture_message
-    alias :annotateException :annotate_exception
-    alias :annotate :annotate_exception
-
     private
 
     def install_at_exit_hook(options)
```
Comment (phillbaker) [inline lib/raven/base.rb]: Whoops, this was moved to Instance, should have cleaned it up here.

File: lib/raven/instance.rb (hunk 3 above)
Comment (phillbaker) [inline lib/raven/instance.rb]: Ah, sorry, this was an artifact of the 2 step process. In the 1st step, I had to remove the `install_at_exit_hook` during the copy to get Instance to pass the specs, then made it match the initial implementation when Base delegated to it.

**Comment by phillbaker** [comment]
@nateberkopec just double checking I understand - I think `Raven.client` will delegate to the instance, it's listed here: https://github.com/phillbaker/raven-ruby/blob/dbb290e9810d5ebf27fabfdb68c047fb7b2fca39/lib/raven/base.rb#L35. I could move that next to the `client=` in the list for clarity
